python/ray/data/_internal/pandas_block.py

```diff
@@ -113,14 +110,20 @@ def __init__(self):
     @staticmethod
     def _table_from_pydict(columns: Dict[str, List[Any]]) -> "pandas.DataFrame":
         pandas = lazy_import_pandas()
-        for key, value in columns.items():
-            if key == TENSOR_COLUMN_NAME or isinstance(
-                next(iter(value), None), np.ndarray
-            ):
+
+        pd_columns: Dict[str, Any] = dict()
+
+        for column_name, column_values in columns.items():
+            np_column_values = convert_to_numpy(column_values)
+
+            if column_name == TENSOR_COLUMN_NAME or np_column_values.ndim > 1:
```

Comment:
Out of scope for this PR, but `TENSOR_COLUMN_NAME` is obsolete and we should remove it